### DISCUSSION TITLE
process communication

### DISCUSSION THREAD

**Discussion by jermatic** [zauberzeug/nicegui#2842] [Ideas / Feature Requests]
Often in our guis for developers when have cpu bound tasks but current there is not easy way for the task to communicate with the UI from a separate process. Using a multiprocessing queue is what I found so far. I did a little digging and found https://pypi.org/project/cengal/ which I think could be used to improve nicegui communication between the ui and the cpu_bound tasks

**Comment by rodja** [comment]
Cengal looks interesting. Do you mean `cengal.hardware.memory.shared_memory` specifically? On the roadmap I saw

> Implement an appropriate Service for cengal.parallel_execution.coroutines - for comfortable shared memory usage inside an async code (including asyncio)

So it could be that shared_memory is not yet working as expected in asnyc applications.
